producer/writer/consumer_service_writer.go

```diff
@@ -132,6 +131,8 @@ func newConsumerServiceWriter(
 		return nil, errUnknownConsumptionType
 	}
 	router := newAckRouter(int(numShards))
+	mPool := newMessagePool(opts.MessagePoolOptions())
```

Comment:
FYI in some cases adding a pool does more harm than good to performance due to the overhead of lock contention trumping the benefit of reduced GC. Not necessarily true for your case, but I've observed that for the samples on the agg tier (you need one sample per timer value), so something that's worth keeping in mind.

Reply:
Yeah, I do notice the lock contention in the pool here is heavy as well, but it still saves tons of allocation and gc in my case, I feel one pool per consumer service is kinda ok for now, in worst case I could try one pool per message writer, but I'm not gonna go that far right now...

Reply:
Have you tried disabling the pool completely and observe the CPU impact? Again it may very well be the case that pooling still does you more good than harm, but I'm just curious.